### ISSUE TITLE
``series.___delitem__`` doesn`t work correctly at MODIN_STORAGE_FORMAT=omnisci

### ISSUE THREAD

**Issue by Rubtsowa** [modin-project/modin#3916]
When I run:
```
import modin.pandas as pd
from modin.pandas.test.utils import test_data
import modin.config as cfg
cfg.StorageFormat.put('omnisci')

data = test_data["float_nan_data"]
modin_series = pd.Series(data[next(iter(data.keys()))])
del modin_series[modin_series.index[0]]
```
I have error:
```
AssertionError: Only column drop is supported
```

**Comment by AndreyPavlenko** [comment]
This issue should be fixed by #3928 and #3925.

**Comment by YarShev** [comment]
Please link this issue to the PR which is going to resolve it.
